File: src/dbflute/adf/meta/summaryWithdrawaldbm.go
package meta

import (
	"github.com/mikeshimura/dbflute/df"
)

type SummaryWithdrawalDbm_T struct {
	df.BaseDBMeta
	ColumnMemberId *df.ColumnInfo
	ColumnMemberName *df.ColumnInfo
	ColumnWithdrawalReasonCode *df.ColumnInfo
	ColumnWithdrawalReasonText *df.ColumnInfo
	ColumnWithdrawalReasonInputText *df.ColumnInfo
	ColumnWithdrawalDatetime *df.ColumnInfo
	ColumnMemberStatusCode *df.ColumnInfo
	ColumnMemberStatusName *df.ColumnInfo
	ColumnMaxPurchasePrice *df.ColumnInfo
}

func (b *SummaryWithdrawalDbm_T) GetProjectName() string {
	return df.DBCurrent_I.ProjectName
}

func (b *SummaryWithdrawalDbm_T) GetDbCurrent() *df.DBCurrent {
	return df.DBCurrent_I
}

var SummaryWithdrawalDbm *SummaryWithdrawalDbm_T

func Create_SummaryWithdrawalDbm() {
	SummaryWithdrawalDbm = new(SummaryWithdrawalDbm_T)
	SummaryWithdrawalDbm.TableDbName = "summary_withdrawal"
	SummaryWithdrawalDbm.TableDispName = "summary_withdrawal"
	SummaryWithdrawalDbm.TablePropertyName = "summaryWithdrawal"
	SummaryWithdrawalDbm.TableSqlName = new(df.TableSqlName)
	SummaryWithdrawalDbm.TableSqlName.TableSqlName = "summary_withdrawal"
	SummaryWithdrawalDbm.TableSqlName.CorrespondingDbName = SummaryWithdrawalDbm.TableDbName

	var summaryWithdrawal df.DBMeta
	summaryWithdrawal = SummaryWithdrawalDbm
	SummaryWithdrawalDbm.DBMeta=&summaryWithdrawal
	memberIdSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo MEMBER_ID
	memberIdSqlName.ColumnSqlName = "MEMBER_ID"
	memberIdSqlName.IrregularChar = false
	SummaryWithdrawalDbm.ColumnMemberId = df.CCI(&summaryWithdrawal, "MEMBER_ID", memberIdSqlName, "", "", "Integer.class", "memberId", "", false, false,true, "INT", 10, 0, "",false,"","", "","","",false,"int64")
	memberNameSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo MEMBER_NAME
	memberNameSqlName.ColumnSqlName = "MEMBER_NAME"
	memberNameSqlName.IrregularChar = false
	SummaryWithdrawalDbm.ColumnMemberName = df.CCI(&summaryWithdrawal, "MEMBER_NAME", memberNameSqlName, "", "", "String.class", "memberName", "", false, false,false, "VARCHAR", 180, 0, "",false,"","", "","","",false,"sql.NullString")
	withdrawalReasonCodeSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo WITHDRAWAL_REASON_CODE
	withdrawalReasonCodeSqlName.ColumnSqlName = "WITHDRAWAL_REASON_CODE"
	withdrawalReasonCodeSqlName.IrregularChar = false
	SummaryWithdrawalDbm.ColumnWithdrawalReasonCode = df.CCI(&summaryWithdrawal, "WITHDRAWAL_REASON_CODE", withdrawalReasonCodeSqlName, "", "", "String.class", "withdrawalReasonCode", "", false, false,false, "CHAR", 3, 0, "",false,"","", "","","",false,"sql.NullString")
	withdrawalReasonTextSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo WITHDRAWAL_REASON_TEXT
	withdrawalReasonTextSqlName.ColumnSqlName = "WITHDRAWAL_REASON_TEXT"
	withdrawalReasonTextSqlName.IrregularChar = false
	SummaryWithdrawalDbm.ColumnWithdrawalReasonText = df.CCI(&summaryWithdrawal, "WITHDRAWAL_REASON_TEXT", withdrawalReasonTextSqlName, "", "", "String.class", "withdrawalReasonText", "", false, false,false, "TEXT", 65535, 0, "",false,"","", "","","",false,"sql.NullString")
	withdrawalReasonInputTextSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo WITHDRAWAL_REASON_INPUT_TEXT
	withdrawalReasonInputTextSqlName.ColumnSqlName = "WITHDRAWAL_REASON_INPUT_TEXT"
	withdrawalReasonInputTextSqlName.IrregularChar = false
	SummaryWithdrawalDbm.ColumnWithdrawalReasonInputText = df.CCI(&summaryWithdrawal, "WITHDRAWAL_REASON_INPUT_TEXT", withdrawalReasonInputTextSqlName, "", "", "String.class", "withdrawalReasonInputText", "", false, false,false, "TEXT", 65535, 0, "",false,"","", "","","",false,"sql.NullString")
	withdrawalDatetimeSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo WITHDRAWAL_DATETIME
	withdrawalDatetimeSqlName.ColumnSqlName = "WITHDRAWAL_DATETIME"
	withdrawalDatetimeSqlName.IrregularChar = false
	SummaryWithdrawalDbm.ColumnWithdrawalDatetime = df.CCI(&summaryWithdrawal, "WITHDRAWAL_DATETIME", withdrawalDatetimeSqlName, "", "", "java.time.LocalDateTime.class", "withdrawalDatetime", "", false, false,true, "DATETIME", 19, 0, "",false,"","", "","","",false,"df.MysqlTimestamp")
	memberStatusCodeSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo MEMBER_STATUS_CODE
	memberStatusCodeSqlName.ColumnSqlName = "MEMBER_STATUS_CODE"
	memberStatusCodeSqlName.IrregularChar = false
	SummaryWithdrawalDbm.ColumnMemberStatusCode = df.CCI(&summaryWithdrawal, "MEMBER_STATUS_CODE", memberStatusCodeSqlName, "", "", "String.class", "memberStatusCode", "", false, false,false, "CHAR", 3, 0, "",false,"","", "","","",false,"sql.NullString")
	memberStatusNameSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo MEMBER_STATUS_NAME
	memberStatusNameSqlName.ColumnSqlName = "MEMBER_STATUS_NAME"
	memberStatusNameSqlName.IrregularChar = false
	SummaryWithdrawalDbm.ColumnMemberStatusName = df.CCI(&summaryWithdrawal, "MEMBER_STATUS_NAME", memberStatusNameSqlName, "", "", "String.class", "memberStatusName", "", false, false,false, "VARCHAR", 50, 0, "",false,"","", "","","",false,"sql.NullString")
	maxPurchasePriceSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo MAX_PURCHASE_PRICE
	maxPurchasePriceSqlName.ColumnSqlName = "MAX_PURCHASE_PRICE"
	maxPurchasePriceSqlName.IrregularChar = false
	SummaryWithdrawalDbm.ColumnMaxPurchasePrice = df.CCI(&summaryWithdrawal, "MAX_PURCHASE_PRICE", maxPurchasePriceSqlName, "", "", "Long.class", "maxPurchasePrice", "", false, false,false, "BIGINT", 19, 0, "",false,"","", "","","",false,"sql.NullInt64")

	SummaryWithdrawalDbm.ColumnInfoList = new(df.List)
	SummaryWithdrawalDbm.ColumnInfoList.Add(SummaryWithdrawalDbm.ColumnMemberId)
	SummaryWithdrawalDbm.ColumnInfoList.Add(SummaryWithdrawalDbm.ColumnMemberName)
	SummaryWithdrawalDbm.ColumnInfoList.Add(SummaryWithdrawalDbm.ColumnWithdrawalReasonCode)
	SummaryWithdrawalDbm.ColumnInfoList.Add(SummaryWithdrawalDbm.ColumnWithdrawalReasonText)
	SummaryWithdrawalDbm.ColumnInfoList.Add(SummaryWithdrawalDbm.ColumnWithdrawalReasonInputText)
	SummaryWithdrawalDbm.ColumnInfoList.Add(SummaryWithdrawalDbm.ColumnWithdrawalDatetime)
	SummaryWithdrawalDbm.ColumnInfoList.Add(SummaryWithdrawalDbm.ColumnMemberStatusCode)
	SummaryWithdrawalDbm.ColumnInfoList.Add(SummaryWithdrawalDbm.ColumnMemberStatusName)
	SummaryWithdrawalDbm.ColumnInfoList.Add(SummaryWithdrawalDbm.ColumnMaxPurchasePrice)


	SummaryWithdrawalDbm.ColumnInfoMap=make(map[string]int)
	SummaryWithdrawalDbm.ColumnInfoMap["memberId"]=0
		SummaryWithdrawalDbm.ColumnInfoMap["memberName"]=1
		SummaryWithdrawalDbm.ColumnInfoMap["withdrawalReasonCode"]=2
		SummaryWithdrawalDbm.ColumnInfoMap["withdrawalReasonText"]=3
		SummaryWithdrawalDbm.ColumnInfoMap["withdrawalReasonInputText"]=4
		SummaryWithdrawalDbm.ColumnInfoMap["withdrawalDatetime"]=5
		SummaryWithdrawalDbm.ColumnInfoMap["memberStatusCode"]=6
		SummaryWithdrawalDbm.ColumnInfoMap["memberStatusName"]=7
		SummaryWithdrawalDbm.ColumnInfoMap["maxPurchasePrice"]=8
	    SummaryWithdrawalDbm.PrimaryKey = false
    SummaryWithdrawalDbm.CompoundPrimaryKey = false

	var summaryWithdrawalMeta df.DBMeta = SummaryWithdrawalDbm
	df.DBMetaInstanceHandler_I.TableDbNameInstanceMap["SummaryWithdrawal"] = &summaryWithdrawalMeta
}

File: src/main/java/com/mssoftech/dbflute/bsentity/dbmeta/SummaryWithdrawalDbm.java
package com.mssoftech.dbflute.bsentity.dbmeta;

import java.util.List;
import java.util.Map;

import org.dbflute.Entity;
import org.dbflute.dbmeta.AbstractDBMeta;
import org.dbflute.dbmeta.info.*;
import org.dbflute.dbmeta.name.*;
import org.dbflute.dbmeta.property.PropertyGateway;
import org.dbflute.dbway.DBDef;
import com.mssoftech.dbflute.allcommon.*;
import com.mssoftech.dbflute.exentity.*;

/**
 * The DB meta of summary_withdrawal. (Singleton)
 * @author DBFlute(AutoGenerator)
 */
public class SummaryWithdrawalDbm extends AbstractDBMeta {

    // ===================================================================================
    //                                                                           Singleton
    //                                                                           =========
    private static final SummaryWithdrawalDbm _instance = new SummaryWithdrawalDbm();
    private SummaryWithdrawalDbm() {}
    public static SummaryWithdrawalDbm getInstance() { return _instance; }

    // ===================================================================================
    //                                                                       Current DBDef
    //                                                                       =============
    public String getProjectName() { return DBCurrent.getInstance().projectName(); }
    public String getProjectPrefix() { return DBCurrent.getInstance().projectPrefix(); }
    public String getGenerationGapBasePrefix() { return DBCurrent.getInstance().generationGapBasePrefix(); }
    public DBDef getCurrentDBDef() { return DBCurrent.getInstance().currentDBDef(); }

    // ===================================================================================
    //                                                                    Property Gateway
    //                                                                    ================
    // -----------------------------------------------------
    //                                       Column Property
    //                                       ---------------
    protected final Map<String, PropertyGateway> _epgMap = newHashMap();
    { xsetupEpg(); }
    protected void xsetupEpg() {
        setupEpg(_epgMap, et -> ((SummaryWithdrawal)et).getMemberId(), (et, vl) -> ((SummaryWithdrawal)et).setMemberId(cti(vl)), "memberId");
        setupEpg(_epgMap, et -> ((SummaryWithdrawal)et).getMemberName(), (et, vl) -> ((SummaryWithdrawal)et).setMemberName((String)vl), "memberName");
        setupEpg(_epgMap, et -> ((SummaryWithdrawal)et).getWithdrawalReasonCode(), (et, vl) -> ((SummaryWithdrawal)et).setWithdrawalReasonCode((String)vl), "withdrawalReasonCode");
        setupEpg(_epgMap, et -> ((SummaryWithdrawal)et).getWithdrawalReasonText(), (et, vl) -> ((SummaryWithdrawal)et).setWithdrawalReasonText((String)vl), "withdrawalReasonText");
        setupEpg(_epgMap, et -> ((SummaryWithdrawal)et).getWithdrawalReasonInputText(), (et, vl) -> ((SummaryWithdrawal)et).setWithdrawalReasonInputText((String)vl), "withdrawalReasonInputText");
        setupEpg(_epgMap, et -> ((SummaryWithdrawal)et).getWithdrawalDatetime(), (et, vl) -> ((SummaryWithdrawal)et).setWithdrawalDatetime(ctldt(vl)), "withdrawalDatetime");
        setupEpg(_epgMap, et -> ((SummaryWithdrawal)et).getMemberStatusCode(), (et, vl) -> ((SummaryWithdrawal)et).setMemberStatusCode((String)vl), "memberStatusCode");
        setupEpg(_epgMap, et -> ((SummaryWithdrawal)et).getMemberStatusName(), (et, vl) -> ((SummaryWithdrawal)et).setMemberStatusName((String)vl), "memberStatusName");
        setupEpg(_epgMap, et -> ((SummaryWithdrawal)et).getMaxPurchasePrice(), (et, vl) -> ((SummaryWithdrawal)et).setMaxPurchasePrice(ctl(vl)), "maxPurchasePrice");
    }
    public PropertyGateway findPropertyGateway(String prop)
    { return doFindEpg(_epgMap, prop); }

    // ===================================================================================
    //                                                                          Table Info
    //                                                                          ==========
    protected final String _tableDbName = "summary_withdrawal";
    protected final String _tableDispName = "summary_withdrawal";
    protected final String _tablePropertyName = "summaryWithdrawal";
    protected final TableSqlName _tableSqlName = new TableSqlName("summary_withdrawal", _tableDbName);
    { _tableSqlName.xacceptFilter(DBFluteConfig.getInstance().getTableSqlNameFilter()); }
    public String getTableDbName() { return _tableDbName; }
    public String getTableDispName() { return _tableDispName; }
    public String getTablePropertyName() { return _tablePropertyName; }
    public TableSqlName getTableSqlName() { return _tableSqlName; }

    // ===================================================================================
    //                                                                         Column Info
    //                                                                         ===========
    protected final ColumnInfo _columnMemberId = cci("MEMBER_ID", "MEMBER_ID", null, null, Integer.class, "memberId", null, false, false, true, "INT", 10, 0, null, false, null, null, null, null, null, false);
//"int64"
    protected final ColumnInfo _columnMemberName = cci("MEMBER_NAME", "MEMBER_NAME", null, null, String.class, "memberName", null, false, false, false, "VARCHAR", 180, 0, null, false, null, null, null, null, null, false);
//"sql.NullString"
    protected final ColumnInfo _columnWithdrawalReasonCode = cci("WITHDRAWAL_REASON_CODE", "WITHDRAWAL_REASON_CODE", null, null, String.class, "withdrawalReasonCode", null, false, false, false, "CHAR", 3, 0, null, false, null, null, null, null, null, false);
//"sql.NullString"
    protected final ColumnInfo _columnWithdrawalReasonText = cci("WITHDRAWAL_REASON_TEXT", "WITHDRAWAL_REASON_TEXT", null, null, String.class, "withdrawalReasonText", null, false, false, false, "TEXT", 65535, 0, null, false, null, null, null, null, null, false);
//"sql.NullString"
    protected final ColumnInfo _columnWithdrawalReasonInputText = cci("WITHDRAWAL_REASON_INPUT_TEXT", "WITHDRAWAL_REASON_INPUT_TEXT", null, null, String.class, "withdrawalReasonInputText", null, false, false, false, "TEXT", 65535, 0, null, false, null, null, null, null, null, false);
//"sql.NullString"
    protected final ColumnInfo _columnWithdrawalDatetime = cci("WITHDRAWAL_DATETIME", "WITHDRAWAL_DATETIME", null, null, java.time.LocalDateTime.class, "withdrawalDatetime", null, false, false, true, "DATETIME", 19, 0, null, false, null, null, null, null, null, false);
//"df.MysqlTimestamp"
    protected final ColumnInfo _columnMemberStatusCode = cci("MEMBER_STATUS_CODE", "MEMBER_STATUS_CODE", null, null, String.class, "memberStatusCode", null, false, false, false, "CHAR", 3, 0, null, false, null, null, null, null, null, false);
//"sql.NullString"
    protected final ColumnInfo _columnMemberStatusName = cci("MEMBER_STATUS_NAME", "MEMBER_STATUS_NAME", null, null, String.class, "memberStatusName", null, false, false, false, "VARCHAR", 50, 0, null, false, null, null, null, null, null, false);
//"sql.NullString"
    protected final ColumnInfo _columnMaxPurchasePrice = cci("MAX_PURCHASE_PRICE", "MAX_PURCHASE_PRICE", null, null, Long.class, "maxPurchasePrice", null, false, false, false, "BIGINT", 19, 0, null, false, null, null, null, null, null, false);
//"sql.NullInt64"

    /**
     * MEMBER_ID: {NotNull, INT(10)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnMemberId() { return _columnMemberId; }
    /**
     * MEMBER_NAME: {VARCHAR(180)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnMemberName() { return _columnMemberName; }
    /**
     * WITHDRAWAL_REASON_CODE: {CHAR(3)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnWithdrawalReasonCode() { return _columnWithdrawalReasonCode; }
    /**
     * WITHDRAWAL_REASON_TEXT: {TEXT(65535)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnWithdrawalReasonText() { return _columnWithdrawalReasonText; }
    /**
     * WITHDRAWAL_REASON_INPUT_TEXT: {TEXT(65535)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnWithdrawalReasonInputText() { return _columnWithdrawalReasonInputText; }
    /**
     * WITHDRAWAL_DATETIME: {NotNull, DATETIME(19)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnWithdrawalDatetime() { return _columnWithdrawalDatetime; }
    /**
     * MEMBER_STATUS_CODE: {CHAR(3)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnMemberStatusCode() { return _columnMemberStatusCode; }
    /**
     * MEMBER_STATUS_NAME: {VARCHAR(50)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnMemberStatusName() { return _columnMemberStatusName; }
    /**
     * MAX_PURCHASE_PRICE: {BIGINT(19)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnMaxPurchasePrice() { return _columnMaxPurchasePrice; }

    protected List<ColumnInfo> ccil() {
        List<ColumnInfo> ls = newArrayList();
        ls.add(columnMemberId());
        ls.add(columnMemberName());
        ls.add(columnWithdrawalReasonCode());
        ls.add(columnWithdrawalReasonText());
        ls.add(columnWithdrawalReasonInputText());
        ls.add(columnWithdrawalDatetime());
        ls.add(columnMemberStatusCode());
        ls.add(columnMemberStatusName());
        ls.add(columnMaxPurchasePrice());
        return ls;
    }

    { initializeInformationResource(); }

    // ===================================================================================
    //                                                                         Unique Info
    //                                                                         ===========
    // -----------------------------------------------------
    //                                       Primary Element
    //                                       ---------------
    protected UniqueInfo cpui() {
        throw new UnsupportedOperationException("The table does not have primary key: " + getTableDbName());
    }
    public boolean hasPrimaryKey() { return false; }
    public boolean hasCompoundPrimaryKey() { return false; }

    // ===================================================================================
    //                                                                       Relation Info
    //                                                                       =============
    // cannot cache because it uses related DB meta instance while booting
    // (instead, cached by super's collection)
    // -----------------------------------------------------
    //                                      Foreign Property
    //                                      ----------------

    // -----------------------------------------------------
    //                                     Referrer Property
    //                                     -----------------

    // ===================================================================================
    //                                                                        Various Info
    //                                                                        ============

    // ===================================================================================
    //                                                                           Type Name
    //                                                                           =========
    public String getEntityTypeName() { return "com.mssoftech.dbflute.exentity.SummaryWithdrawal"; }
    public String getConditionBeanTypeName() { return "com.mssoftech.dbflute.cbean.SummaryWithdrawalCB"; }
    public String getBehaviorTypeName() { return "com.mssoftech.dbflute.exbhv.SummaryWithdrawalBhv"; }

    // ===================================================================================
    //                                                                         Object Type
    //                                                                         ===========
    public Class<SummaryWithdrawal> getEntityType() { return SummaryWithdrawal.class; }

    // ===================================================================================
    //                                                                     Object Instance
    //                                                                     ===============
    public SummaryWithdrawal newEntity() { return new SummaryWithdrawal(); }

    // ===================================================================================
    //                                                                   Map Communication
    //                                                                   =================
    public void acceptPrimaryKeyMap(Entity et, Map<String, ? extends Object> mp)
    { doAcceptPrimaryKeyMap((SummaryWithdrawal)et, mp); }
    public void acceptAllColumnMap(Entity et, Map<String, ? extends Object> mp)
    { doAcceptAllColumnMap((SummaryWithdrawal)et, mp); }
    public Map<String, Object> extractPrimaryKeyMap(Entity et) { return doExtractPrimaryKeyMap(et); }
    public Map<String, Object> extractAllColumnMap(Entity et) { return doExtractAllColumnMap(et); }
}
